File: java/src/main/java/io/fivexer/sdk/model/UpsertSkill.java
package io.fivexer.sdk.model;

import io.fivexer.sdk.internal.Json;

/**
 * Input for defining a skill.
 *
 * <p>Fluent builder: set what you need, then hand it to the client. {@link #toJson()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
public class UpsertSkill {
    private String key;
    private String name;
    private String description;

    public UpsertSkill(String key, String name) {
        if (key == null) {
            throw new IllegalArgumentException("key is required");
        }
        if (name == null) {
            throw new IllegalArgumentException("name is required");
        }
        this.key = key;
        this.name = name;
    }

    public String getKey() { return key; }
    public String getName() { return name; }
    public String getDescription() { return description; }

    public UpsertSkill description(String description) { this.description = description; return this; }

    /** Serialise for the wire, omitting unset fields. */
    public String toJson() {
        return Json.write(this);
    }
}

File: php/src/Model/UpsertSkill.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * Input for defining a skill.
 *
 * Fluent builder: set what you need, then hand it to the client. {@see self::toArray()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
final class UpsertSkill
{
    public function __construct(
        private readonly string $key,
        private readonly string $name,
        private ?string $description = null,
    ) {
    }

    public function description(string $description): self
    {
        $this->description = $description;
        return $this;
    }

    /**
     * Serialise for the wire, omitting unset fields.
     *
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'key' => $this->key,
            'name' => $this->name,
        ] + Json::compact([
            'description' => $this->description,
        ]);
    }
}
